evil-ex.el

```diff
@@ -40,6 +40,7 @@
 ;; substitution command (":s/foo/bar"), the handler incrementally
 ;; highlights matches in the buffer as the substitution is typed.
 
+(require 'comint)
```

Comment:
Please remove the `(require 'comint nil t)` further below.

Reply:
That being said, you could give the `(require 'shell nil t)` the same treatment and add `(require 'shell)` at the top after removing it.

Reply:
Adding ``(require comint)`` was to eliminate the

    Warning: the function ‘comint-completion-at-point’ is not known to be defined.

byte compiler warning message. 

The reason for requiring ``comint`` was because function ``comint-completion-at-point`` is defined in this file. ``(require shell)`` works too because ``shell.el`` included ``comint.el``. However, I personally prefer requiring the smallest possible dependency. If you don't agree with me, feel free to change to ``(require shell)``.

Reply:
That's fine, my top concern is making sure there are no duplicate imports which would confuse a future reader.  With this change of yours, both `(require 'shell nil t)` and `(require 'comint nil t)` have become unnecessary and should be removed as well.

Reply:
No, all my changes address different byte compilation warnings. `(require comint)` targets the warning:

    Warning: the function ‘comint-completion-at-point’ is not known to be defined

You may need to run `make clean` before running `make` to see the previous warning messages.

Reply:
I'm aware of that, just pointing out that `comint` is required twice (and `shell` once). You don't need that second import.

Reply:
Yes, you are right, we don't need to require `comint` elsewhere in this file. My apology for missing your point. I've deleted the two function level `require`.

Reply:
I've also changed to requiring `shell` instead of `comint` as per your suggestion. Thanks.